server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileChangelogAction.java

```diff
@@ -0,0 +1,142 @@
+    this.service = service;
+    this.dbClient = dbClient;
+    this.profileFactory = profileFactory;
+    this.languages = languages;
+  }
+
+  @Override
+  public void define(NewController context) {
+    NewAction changelog = context.createAction("changelog")
+      .setSince("5.2")
+      .setDescription("Get the history of changes on a quality profile: rule activation/deactivation, change in parameters/severity. " +
+        "Events are ordered by date in descending order (most recent first).")
+      .setHandler(this)
+      .setResponseExample(getClass().getResource("example-changelog.json"));
+
+    QProfileIdentificationParamUtils.defineProfileParams(changelog, languages);
+
+    changelog.addPagingParams(50);
+
+    changelog.createParam(PARAM_SINCE)
+      .setDescription("Start date for the changelog.")
+      .setExampleValue("2011-04-25");
+
+    changelog.createParam(PARAM_TO)
+      .setDescription("End date for the changelog.")
+      .setExampleValue("2013-07-25");
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    DbSession session = dbClient.openSession(false);
+    try {
+      String profileKey = QProfileIdentificationParamUtils.getProfileKeyFromParameters(request, profileFactory, session);
+      if (dbClient.qualityProfileDao().getByKey(session, profileKey) == null) {
+        throw new NotFoundException(String.format("Could not find a profile with key '%s'", profileKey));
+      }
+
+      QProfileActivityQuery query = new QProfileActivityQuery().setQprofileKey(profileKey);
+      Date since = request.paramAsDate(PARAM_SINCE);
+      if (since != null) {
+        query.setSince(since);
+      }
+      Date to = request.paramAsDate(PARAM_TO);
+      if (to != null) {
+        query.setTo(to);
+      }
+      SearchOptions options = new SearchOptions();
+
+      int page = request.mandatoryParamAsInt(Param.PAGE);
+      options.setPage(page, request.mandatoryParamAsInt(Param.PAGE_SIZE));
+
+      Result<QProfileActivity> result = service.searchActivities(query, options);
+      writeResponse(response.newJsonWriter(), result, Paging.create(options.getLimit(), page, (int) result.getTotal()));
+    } finally {
+      session.close();
+    }
+  }
+
+  private void writeResponse(JsonWriter json, Result<QProfileActivity> result, Paging paging) {
+    json.beginObject();
+    json.prop("total", result.getTotal());
+    json.prop(Param.PAGE, paging.pageIndex());
+    json.prop(Param.PAGE_SIZE, paging.pageSize());
+    json.name("events").beginArray();
+    for (QProfileActivity event : result.getHits()) {
+      json.beginObject()
+        .prop("date", DateUtils.formatDateTime(event.getCreatedAt()))
+        .prop("authorLogin", event.getLogin())
+        .prop("authorName", event.authorName())
+        .prop("action", event.getAction())
+        .prop("ruleKey", event.ruleKey().toString())
+        .prop("ruleName", event.ruleName());
```

Comment:
Open question : should we repeat each time the authorName and ruleName, or should we set them in separate repositories (like /api/issues/search) ?